client/blocks/video-editor/index.jsx

```diff
@@ -0,0 +1,230 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import ProgressBar from 'components/progress-bar';
+import Notice from 'components/notice';
+import DetailPreviewVideo from 'post-editor/media-modal/detail/detail-preview-video';
+import VideoEditorButtons from './video-editor-buttons';
+import { updatePoster } from 'state/ui/editor/video-editor/actions';
+import {
+	getPosterUploadProgress,
+	getPosterUrl,
+	shouldCloseVideoEditorModal,
+	shouldShowVideoEditorError,
+} from 'state/selectors';
+
+/**
+ * Module variables
+ */
+let isSelectingFrame = false;
+
+class VideoEditor extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		media: PropTypes.object.isRequired,
+		onCancel: PropTypes.func,
+		onUpdatePoster: PropTypes.func,
+
+		// Connected props
+		posterUrl: PropTypes.string,
+		shouldCloseModal: PropTypes.bool,
+		shouldShowError: PropTypes.bool,
+		uploadProgress: PropTypes.number,
+	};
+
+	static defaultProps = {
+		onCancel: noop,
+		onUpdatePoster: noop,
+	};
+
+	state = {
+		error: false,
+		isLoading: true,
+		pauseVideo: false,
+	};
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! nextProps.shouldCloseModal && ! nextProps.shouldShowError ) {
+			return;
+		}
+
+		if ( nextProps.shouldShowError ) {
```

Comment:
That's because `shouldShowError` only indicates a problem with trying to update the poster, but there are other types of errors (e.g. [error loading the VideoPress script](https://github.com/Automattic/wp-calypso/pull/11648/files#diff-35b626cb9b7af72412a7ab302d8fc3f6R185)). They also result in [rendering the same error](https://github.com/Automattic/wp-calypso/pull/11648/files#diff-35b626cb9b7af72412a7ab302d8fc3f6R138) to the user though.

Do you think the naming could be improved somewhere to reduce confusion?